src/librustc_mir/transform/const_prop.rs

```diff
@@ -335,6 +348,16 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
     }
 
     fn get_const(&self, local: Local) -> Option<Const<'tcx>> {
+        if local == RETURN_PLACE {
+            // Try to read the return place as an immediate so that if it is representable as a
+            // scalar, we can handle it as such, but otherwise, just return the value as is.
+            if let Some(Ok(Ok(imm))) = self.ret.map(|ret| self.ecx.try_read_immediate(ret)) {
```

Comment:
Consider `return match self.ret.map(...) {` instead here; seems simpler?

Reply:
As in:

```rust
return match self.ret.map(|ret| self.ecx.try_read_immediate(ret)) {
    Some(Ok(Ok(imm))) => Some(imm.into()),
    _ => self.ret,
};
```

Reply:
Yep :)